app/styles/app.css

```diff
@@ -211,6 +211,69 @@ nav a:hover {
     padding-right: 1em;
 }
 
+.SearchableMetricSelector {
+    display: inline-block;
+    padding: 0.5em 0;
+    padding-right: 1em;
+}
+
+.SearchableMetricSelector .SelectorModal {
+    left: 0;
+    right: 0;
+    top: 0;
+    bottom: 0;
+    position: fixed;
+    z-index: 1;
+    background-color: rgba(0,0,0,0.5);
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+
+.SearchableMetricSelector .modal-content {
+    width: 500px;
+    background-color: white;
+}
+
+.SearchableMetricSelector .modal-body {
+    border-top: 1px solid #eee;
+    border-bottom: 1px solid #eee;
+    max-height: calc(100vh - 210px);
+    overflow-y: auto;
+}
+
+.SearchableMetricSelector .modal-header, 
+.SearchableMetricSelector .modal-footer {
+    padding: 10px
+}
+
+.SearchableMetricSelector .selectorButton {
+    width: 500px;
+    text-align: left;
+    text-overflow: ellipsis;
+    overflow: hidden;
+    white-space: nowrap;
+    
+}
+
+@media screen and (max-width: 500px) {
+    
+    .SearchableMetricSelector .selectorButton{
+        width: calc(100vw - 4em);
+        display: block;
+    }
+    
+    .SearchableMetricSelector .modal-content {
+        width:100%;
+    }
+    
+    .SearchableMetricSelector .selectorLabel{
```

Comment:
Only piece of critical feedback is to pick one format of class name and be consistent, at least within one component. Here you have modal-content, selectorLabel, and SearchableMetricSelector all as different ways of writing a classname.